src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineViewModel.cs

```diff
@@ -262,6 +292,13 @@ private async ValueTask ComputeViewStateAsync(CancellationToken cancellationToke
             if (_isDisposed)
                 return;
 
+            if (Visibility != Visibility.Visible)
+            {
+                // Retry the update after a delay
+                _workQueue.AddWork(cancelExistingWork: true);
+                return;
+            }
```

Comment:
instead of all 3, it likely suffices to just have teh first case.  worst that happens is that the user switches away, and we finish up the work here.  but any future work requsts will bail out at the top.

Reply:
Sure, that works for me

Reply:
On second thought, given the difficulty of proving the change works, I'm concerned about edge cases like switching back and forth between a Windows Forms designer and a C# source document (resulting in a visibility change for each switch). I'm more comfortable with the original change in terms of confidence that work is avoided when the UI is not visible. If you'd like, I could eliminate the _middle_ of the three checks.